content/05.main-text.md

```diff
@@ -34,6 +55,38 @@ Assess ability to predict paths in <https://github.com/SuLab/DrugMechDB>
 
 ## Methods {.page_break_before}
 
+### The HetMat awakens
+
+At the core of the hetmatpy package is the HetMat data structure for storing and accessing the network.
+HetMats are stored on disk as a directory, which by convention uses a `.hetmat` extension.
+A HetMat directory stores a single heterogeneous network, whose data resides in the following files.
+
+1. A `metagraph.json` file stores the schema, defining which types of nodes and edges comprise the hetnet.
+   This format is defined by the [hetnetpy](https://github.com/hetio/hetnetpy) Python package.
+   Hetnetpy was originally developed with the name hetio during prior studies
+   [@hetio-dag; @rephetio],
+   but we [renamed](https://github.com/hetio/hetnetpy/issues/40) it to hetnetpy for better disambiguation.
+2. A `nodes` directory containing one file per node type (metanode) that defines each node.
+   Currently, `.tsv` files are supported where each row represents a node.
+3. An `edges` directory containing one file per edge type (metadata) that encodes the adjacency matrix. 
+   The matrix can be serialized using either the Numpy dense format (`.npy`) or SciPy sparse format (`.sparse.npz`).
+
+For node and edge files, compression is supported as detected from `.gz`, `.bz2`, `.zip`, and `.xz` extensions.
+This structure of storing a hetnet allows for selective reading of nodes and edges into memory.
+For example, a certain computation may only require access to a subset of the node and edge types.
+By only loading the required node and edge types, we reduce memory usage and read times.
+
+Additional subdirectories, such as `path-counts` and `permutations`, store data generated from the HetMat.
+By using consistent paths for generated data, we avoid recomputing data that already exists on disk.
+A HetMat directory can be zipped for archiving and transfer.
+Users can selectively include generated data in archives.
+Since the primary application of HetMats is to generate computationally demanding measurements on hetnets, the ability to share HetMats with precomputed data is paramount.
+
+The [`HetMat`](https://hetio.github.io/hetmatpy/reference/hetmatpy/hetmat/#hetmat) class implements the above logic.
```

Comment:
I wonder if it'd be better to introduce the class first and then say what it includes.

Reply:
the thought here is that the data structure is a bit more generic than the specific python handler class.

one could imagine another language or library supports the data structure and on-disk format.